### PR TITLE
DhtArc stability 01: initial tests

### PR DESCRIPTION
**Pull request by maackle**
### Summary

Adds some tests which test for arc resizing stability: a basic check of convergence vs. divergence. These all live in `arc_stability.rc`

Mostly other changes are just a bunch of renames from `PeerDensity` to `PeerViewAlpha`, paving the way for having multiple different methods of calculating ideal target coverage.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/kitsune_p2p/dht_arc/src/dht_arc/dht_arc_set.rs [hunk 1]
```diff
@@ -372,26 +372,56 @@ impl ArcInterval<DhtLocation> {
...
+                    } else {
+                        vec![
+                            "-".repeat(hi + 1),
+                            " ".repeat((lo - hi).saturating_sub(1)),
+                            "-".repeat(len - lo),
+                        ]
+                    }
                 }
                 .join("");
                 let center = loc_downscale(len, self.center_loc());
```
Comment (maackle) [inline crates/kitsune_p2p/dht_arc/src/dht_arc/dht_arc_set.rs:427]: Note: this all gets greatly simplified in a subsequent PR so don't worry about it

**Comment by freesig** [comment]
Looks good but need a little more comments to understand what's going on.

File: crates/holochain/tests/dht_arc.rs [hunk 2]
```diff
@@ -40,15 +40,15 @@ async fn test_arc_coverage() {
...
+                let density = bucket.peer_view_alpha();
                 arc.update_length(density);
             }
             let bucket = DhtArcBucket::new(
                 DhtArc::new(MAX_HALF_LENGTH / 2, MAX_HALF_LENGTH),
                 peers.clone(),
             );
             println!("{}", bucket);
-            std::thread::sleep(std::time::Duration::from_millis(200));
+            // std::thread::sleep(std::time::Duration::from_millis(200));
```
Comment (neonphog) [inline crates/holochain/tests/dht_arc.rs:51]:
```suggestion
```

Comment (maackle) [reply crates/holochain/tests/dht_arc.rs:51]: This entire test is removed in a subsequent PR

**Review by neonphog — APPROVED** [review]

**Comment by maackle** [comment]
@freesig want a small tour? There are also some more comments in the subsequent PRs.

File: crates/kitsune_p2p/dht_arc/src/dht_arc/dht_arc_bucket.rs [hunk 3]
```diff
@@ -93,8 +103,8 @@ impl DhtArcBucket {
         }
     }
 
-    /// Get the density of this bucket.
-    pub fn density(&self) -> PeerDensity {
+    #[deprecated = "use peer_view"]
```
Comment (freesig) [inline crates/kitsune_p2p/dht_arc/src/dht_arc/dht_arc_bucket.rs:106]: Why does this PR introduce a deprecated function?
Is this for cases where you don't want people to choose the version of peer view?

**Review by freesig — APPROVED** [review]
Looks good, a few suggestions.

**Pull request merged**